Sources/Vapor/HTTP/Headers/HTTPHeaders+ContentRange.swift

```diff
@@ -0,0 +1,263 @@
+        }
+        
+    }
+    
+    /// Convenience for accessing the Content-Range response header.
+    ///
+    /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range
+    public var contentRange: ContentRange? {
+        get {
+            let directives = parseDirectives(name: .contentRange).flatMap { $0 }
+            return HTTPHeaders.ContentRange(directives: directives)
+        }
+        set {
+            if contains(name: .contentRange) {
+                remove(name: .contentRange)
+            }
+            if newValue == nil {
+                return
+            }
+            add(name: .contentRange, value: newValue!.serialize())
+        }
+    }
+    
+    /// Convenience for accessing the `Range` request header.
+    ///
+    /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range
+    public var range: Range? {
+        get {
+            return HTTPHeaders.Range(directives: self.parseDirectives(name: .range).flatMap { $0 })
+        }
+        set {
+            if contains(name: .range) {
+                remove(name: .range)
+            }
+            if newValue == nil {
+                return
+            }
+            add(name: .range, value: newValue!.serialize())
+        }
+    }
+}
+
+extension HTTPHeaders.Range {
+    /// Represents one value of the `Range` request header.
+    ///
+    /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range
+    public enum Value: Equatable {
+        ///Integer with single trailing dash, e.g. `25-`
+        case start(value: Int)
+        ///Integer with single leading dash, e.g. `-25`
+        case tail(value: Int)
+        ///Two integers with single dash in between, e.g. `20-25`
+        case within(start: Int, end: Int)
+        
+        ///Parses a string representing a requested range in one of the following formats:
+        ///
+        ///- `<range-start>-<range-end>`
+        ///- `-<range-end>`
+        ///- `<range-start>-`
+        ///
+        /// - parameters:
+        ///     - requestStr: String representing a requested range
+        /// - returns: A `HTTPHeaders.Range.Value` if the `requestStr` is valid, `nil` otherwise.
+        public static func from<T>(requestStr: T) -> HTTPHeaders.Range.Value? where T: StringProtocol {
+            let ranges = requestStr.split(separator: "-", omittingEmptySubsequences: false)
+                .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+            guard ranges.count <= 2 else { return nil }
+            let start = Int(ranges[0])
+            let end = Int(ranges.count > 1 ? ranges[1] : "")
+            if start == nil && end == nil {
+                return nil
+            }
+            if start != nil && end == nil {
+                return .start(value: start!)
+            }
+            if start == nil && end != nil {
+                return .tail(value: end!)
+            }
+            return .within(start: start!, end: end!)
```

Comment:
This is a way to avoid force unwraps:
```suggestion
            let count = ranges.count
            guard count <= 2 else { return nil }
    
            switch (count > 0 ? Int(ranges[0]) : nil, count > 1 ? Int(ranges[1]) : nil) {
            case (nil, nil):
                return nil
            case let (.some(start), nil):
                return .start(value: start)
            case let (nil, .some(tail)):
                return .tail(value: tail)
            case let (.some(start), .some(end)):
                return .within(start: start, end: end)
            }
```